sale_commission_formula/models/sale_commission_mixin.py

```diff
@@ -0,0 +1,29 @@
+#  -*- coding: utf-8 -*-
+#  Copyright 2019 Simone Rubino - Agile Business Group
+#  License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, models
+from odoo.tools.safe_eval import safe_eval
+
+
+class SaleCommissionMixin(models.AbstractModel):
+    _inherit = 'sale.commission.mixin'
+
+    @api.model
+    def _get_formula_input_dict(self):
+        return {'line': self.object_id,
+                'self': self}
+
+    @api.multi
+    def _get_commission_amount(
+            self, commission, subtotal, commission_free, product, quantity):
+        self.ensure_one()
+        if commission.commission_type == 'formula' and \
+                not commission_free and commission:
+            formula = commission.formula
+            results = self._get_formula_input_dict()
+            safe_eval(formula, results, mode="exec", nocopy=True)
+            return float(results['result'])
+        else:
```

Comment:
No need of `else`

Reply:
@pedrobaeza I am calling super to get the correct `amount` if the `commission_type` is not `formula`.